components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/NetworkMessageEncoder.cs

```diff
@@ -82,16 +83,18 @@ public class NetworkMessageEncoder : IMessageEncoder {
         /// <summary>
         /// DataSetMessage to NetworkMessage Json batched encoding
         /// </summary>
-        /// <param name="messages"></param>
-        /// <param name="maxMessageSize"></param>
+        /// <param name="messages">Messages to encode</param>
+        /// <param name="maxMessageSize">Maximum size of messages</param>
+        /// <param name="useReversibleEncoding">Flag for reversible encoding</param>
         /// <returns></returns>
         private IEnumerable<NetworkMessageModel> EncodeBatchAsJson(
-            IEnumerable<DataSetMessageModel> messages, int maxMessageSize) {
+            IEnumerable<DataSetMessageModel> messages, int maxMessageSize, bool useReversibleEncoding) {
 
             // by design all messages are generated in the same session context,
             // therefore it is safe to get the first message's context
             var encodingContext = messages.FirstOrDefault(m => m.ServiceMessageContext != null)
                 ?.ServiceMessageContext;
+            var compressedPayload = messages.Where(x => x.CompressedPayload).Any();
```

Comment:
If we add message payload compression functionality to `NetworkMessageEncoder` then we also need to support it in `MonitoredItemMessageEncoder`. Otherwise the compression parts should be reverted.

Reply:
As discussed with @cristipogacean, I will remove functionality around `CompressedPayload`.

Reply:
Removed in #1696.